data-prepper-api/src/test/java/com/amazon/dataprepper/model/configuration/SinkModelTest.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.model.configuration;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.fasterxml.jackson.dataformat.yaml.YAMLGenerator;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.hasItem;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasKey;
+
+class SinkModelTest {
+    private ObjectMapper objectMapper;
+
+    @BeforeEach
+    void setup() {
+        objectMapper = new ObjectMapper(YAMLFactory.builder()
+                .disable(YAMLGenerator.Feature.WRITE_DOC_START_MARKER)
+                .enable(YAMLGenerator.Feature.USE_PLATFORM_LINE_BREAKS)
+                .build());
+    }
+
+    @Test
+    void deserialize_with_Sink() throws IOException {
+        final InputStream inputStream = this.getClass().getResourceAsStream("sink_plugin.yaml");
+
+        final SinkModel sinkModel = objectMapper.readValue(inputStream, SinkModel.class);
+
+        assertThat(sinkModel, notNullValue());
+        assertThat(sinkModel.getPluginName(), equalTo("customSinkPlugin"));
```

Comment:
lines 47 to 55 could use assertAll() to produce comprehensive failure messages.

Reply:
Thanks for the suggestion. I tried grouping these `assertAll` statements in related groups. I didn't want to put too much in any given one because it could be confusing to see lots of different errors.